pkg/pgmodel/upgrade_tests/upgrade_test.go

```diff
@@ -292,6 +204,114 @@ func TestUpgradeFromPrev(t *testing.T) {
 	}
 }
 
+func withDBAndConnector(t testing.TB, DBName string, connectorImage string,
+	f func(dbContainer testcontainers.Container, dbTmpDir string, connectorHost string, connectorPort nat.Port)) {
+	var err error
+	ctx := context.Background()
+
+	tmpDir, err := testhelpers.TempDir("update_test_out")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	dbContainer, err := testhelpers.StartPGContainerWithImage(ctx, cleanImage, tmpDir, *printLogs)
+	if err != nil {
+		t.Fatal("Error setting up container", err)
+	}
+
+	defer func() {
```

Comment:
This really should be defer testhelpers.StopContainer(dbContainer, *printLogs); It's weird that this calls StopLogProducer in here while the start was in testhelpers.